### ISSUE TITLE
OTP26 adds a module `argparse` to stdlib 

### ISSUE THREAD

**Issue by hanssv** [max-au/argparse#46]
With the addition of `argparse` to `stdlib` in OTP26 packing up an application using this library fails with:
```
===> Error generating release: 
Duplicated modules: 
    argparse specified in stdlib and argparse
```

Not sure what the best solution is - perhaps simply renaming the top level module to `arg_parse` would be a sensible option?!

Running the same application on OTP23-OTP26 having two different versions of the same thing isn't really an appealing option. But once we reach OTP29 we should be able to drop this.

**Comment by hanssv** [comment]
On second thought, probably some rebar3 magic should be enough...

**Comment by hanssv** [comment]
Adding the following into `rebar.config.script` should work:
```
%% OTP26 has `argparse` built in
OTPVersion = list_to_integer(erlang:system_info(otp_release)),
FilterDepsArgparse =
    case OTPVersion > 25 of
       true  -> fun(Deps) -> lists:keydelete(argparse, 1, Deps) end;
       false -> fun(Deps) -> Deps end
    end,

{deps, Deps0} = lists:keyfind(deps, 1, CONFIG),
lists:keyreplace(deps, 1, CONFIG, {deps, FilterDepsArgparse(Deps0)}).
```

and then a similar thing to do the lock file in `rebar.lock.script` if you use lock-files...

**Comment by hanssv** [comment]
Props to @eproxus for getting my mind on the right track.

**Comment by hanssv** [comment]
Oh, and don't forget to patch your .app file and your `relx` config if you happen to use that 🙈

**Comment by hanssv** [comment]
See aeternity/aeternity#4217 for a reasonably complete example on how to achieve this...

**Comment by max-au** [comment]
Nice trick indeed! In OTP 26, `argparse` is basically a part of it (with a few changes and improvements), and I am trying to understand if there is a good way to keep standalone `argparse` support (especially `cli.erl` which is not going to be a part of OTP).

Open to ideas on that one.

**Comment by hanssv** [comment]
Oh, also beware - the OTP implementation is missing the `argparse:format_error/3` function 🙈

**Comment by hanssv** [comment]
In fact the OTP implementation is subtly different in other ways also (error reporting, etc) - so this was probably a futile attempt.

**Comment by max-au** [comment]
After fiddling and mingling, and guessing and groaning "why didn't I name the module `args` in OTP", I came up to the only viable solution. Renamed the module in `argparse`, and released this as version `2.0.0`.

Here is the PR: https://github.com/max-au/argparse/pull/48

I needed this to keep support for old OTP versions in https://github.com/max-au/erlperf

This works surprisingly well, as both `args` and `argparse` can easily co-exist. From now on, when OTP 26 becomes the lowest supported OTP version, I'll just remove `args` and migrate to `argparse`. Hope this also works for you!
